### PR TITLE
mysql create table add table and colunm comment

### PR DESCRIPTION
**Pull request by Caisin**
## New Features

mysql create table add table and colunm comment

## Changes

- [x]

### CODE REVIEW

File: src/table/create.rs [hunk 1]
```diff
@@ -36,23 +37,23 @@ use crate::{
...
+///         r#") COMMENT 'table comment' "#,
 ///     ].join(" ")
 /// );
 /// assert_eq!(
 ///     table.to_string(PostgresQueryBuilder),
 ///     [
 ///         r#"CREATE TABLE IF NOT EXISTS "character" ("#,
 ///             r#""id" serial NOT NULL PRIMARY KEY,"#,
-///             r#""font_size" integer NOT NULL,"#,
+///             r#""font_size" integer NOT NULL ,"#,
```
Comment (billy1624) [inline src/table/create.rs:56]: The extra space comes from 

https://github.com/SeaQL/sea-query/blob/42465d36c5a1895662a33c4ca087347639473632/src/backend/sqlite/table.rs#L24-L25

I'd suggest moving `write!(sql, " ").unwrap();` line into the `prepare_column_spec` method.

File: src/table/create.rs [hunk 2]
```diff
@@ -36,23 +37,23 @@ use crate::{
...
+///             r#"`font_size` int NOT NULL COMMENT 'font size',"#,
 ///             r#"`character` varchar(255) NOT NULL,"#,
 ///             r#"`size_w` int NOT NULL,"#,
 ///             r#"`size_h` int NOT NULL,"#,
 ///             r#"`font_id` int DEFAULT NULL,"#,
 ///             r#"CONSTRAINT `FK_2e303c3a712662f1fc2a4d0aad6`"#,
 ///                 r#"FOREIGN KEY (`font_id`) REFERENCES `font` (`id`)"#,
 ///                 r#"ON DELETE CASCADE ON UPDATE CASCADE"#,
-///         r#")"#,
+///         r#") COMMENT 'table comment' "#,
```
Comment (billy1624) [inline src/table/create.rs:48]: We have a extra space after the closing quote `'`?

**Review by billy1624 — COMMENTED** [review]
Hey @Caisin, thanks for contributing!! Some comments for you :P

File: src/backend/table_builder.rs [hunk 3]
```diff
@@ -106,14 +106,24 @@ pub trait TableBuilder:
                 self.prepare_generated_column(expr, *stored, sql)
             }
             ColumnSpec::Extra(string) => write!(sql, "{string}").unwrap(),
+            ColumnSpec::Comment(comment) => self.column_comment(comment, sql),
         }
     }
 
+    /// column comment
+    fn column_comment(&self, comment: &String, sql: &mut dyn SqlWriter) {
```
Comment (ikrivosheev) [inline src/backend/table_builder.rs:114]: I think, It should be blanked.

**Review by ikrivosheev — COMMENTED** [review]
@Caisin thank you for the PR! Small comments.

File: src/table/create.rs (hunk 2 above)
Comment (Caisin) [reply src/table/create.rs]: done

File: src/backend/table_builder.rs (hunk 3 above)
Comment (Caisin) [reply src/backend/table_builder.rs]: now is published

**Review by ikrivosheev — CHANGES_REQUESTED** [review]
@Caisin can you move part with rbatist into other PR?

**Review by ikrivosheev — APPROVED** [review]
@CaisinThank you! LGTM!

**Comment by ikrivosheev** [comment]
@billy1624 do you want to include it into 0.29 release?)

File: src/table/create.rs [hunk 4]
```diff
@@ -88,6 +89,8 @@ pub struct TableCreateStatement {
     pub(crate) foreign_keys: Vec<ForeignKeyCreateStatement>,
     pub(crate) if_not_exists: bool,
     pub(crate) check: Vec<SimpleExpr>,
+    ///table comment
```
Comment (tyt2y3) [inline src/table/create.rs:92]:
```suggestion
```

File: src/table/create.rs [hunk 5]
```diff
@@ -123,6 +126,14 @@ impl TableCreateStatement {
         self.table = Some(table.into_table_ref());
         self
     }
```
Comment (tyt2y3) [inline src/table/create.rs:128]:
```suggestion
    }

```

File: src/backend/table_builder.rs [hunk 6]
```diff
@@ -106,14 +106,22 @@ pub trait TableBuilder:
...
+    /// column comment
+    fn column_comment(&self, _comment: &str, _sql: &mut dyn SqlWriter) {}
+
     /// The keyword for setting a column to be auto increment.
     fn column_spec_auto_increment_keyword(&self) -> &str;
 
     /// Translate [`TableOpt`] into SQL statement.
     fn prepare_table_opt(&self, create: &TableCreateStatement, sql: &mut dyn SqlWriter) {
+        self.prepare_table_opt_def(create, sql)
+    }
```
Comment (tyt2y3) [inline src/backend/table_builder.rs:122]:
```suggestion
    }

```

File: src/backend/mysql/table.rs [hunk 7]
```diff
@@ -1,6 +1,15 @@
 use super::*;
 
 impl TableBuilder for MysqlQueryBuilder {
+    fn prepare_table_opt(&self, create: &TableCreateStatement, sql: &mut dyn SqlWriter) {
+        // comment
+        if let Some(comment) = &create.comment {
+            // self.quote()
+            write!(sql, " COMMENT '{comment}'").unwrap();
```
Comment (tyt2y3) [inline src/backend/mysql/table.rs:8]: We should escape the comment properly.

File: src/backend/mysql/table.rs [hunk 8]
```diff
@@ -191,4 +200,10 @@ impl TableBuilder for MysqlQueryBuilder {
             self.prepare_table_ref_table_stmt(to_name, sql);
         }
     }
+
+    /// column comment
+    fn column_comment(&self, comment: &str, sql: &mut dyn SqlWriter) {
+        let comment = comment.replace('\'', "\\'");
+        write!(sql, "COMMENT '{comment}'").unwrap()
```
Comment (tyt2y3) [inline src/backend/mysql/table.rs:207]: We have a standard way to escape string

**Review by tyt2y3 — CHANGES_REQUESTED** [review]
Thank you for the contribution

**Review by Caisin — COMMENTED** [review]
done

**Review by tyt2y3 — APPROVED** [review]
Thank you again for your contribution.

File: src/table/create.rs [hunk 9]
```diff
@@ -37,15 +37,15 @@ use crate::{
...
+///             r#"`font_size` int NOT NULL COMMENT 'font\'s size',"#,
 ///             r#"`character` varchar(255) NOT NULL,"#,
 ///             r#"`size_w` int NOT NULL,"#,
 ///             r#"`size_h` int NOT NULL,"#,
 ///             r#"`font_id` int DEFAULT NULL,"#,
 ///             r#"CONSTRAINT `FK_2e303c3a712662f1fc2a4d0aad6`"#,
 ///                 r#"FOREIGN KEY (`font_id`) REFERENCES `font` (`id`)"#,
 ///                 r#"ON DELETE CASCADE ON UPDATE CASCADE"#,
-///         r#") COMMENT 'table comment'"#,
+///         r#") COMMENT 'table\'s comment'"#,
```
Comment (tyt2y3) [inline src/table/create.rs:48]: I like the extra `'`

**Comment by pkjy** [comment]
sea-orm 0.11.2 still not working with comment ,is this feature published now ?

**Comment by tyt2y3** [comment]
No, it's been merged, but not yet released.

**Pull request merged**